### PR TITLE
Pull Request for Issue1780: Implementing Zebra on BioXAS

### PR DESCRIPTION
**Pull request by helfrij**
This pull request handles adding the Zebra as the main control for the scaler and the Ge 32 element detector. It involved making a new trigger source and some small tweaks to the 32 element detector class. A new action was also created to handle ensuring that the 32 element detector is ready for acquisition since it can be slow some times.

### CODE REVIEW

File: source/acquaman/BioXAS/BioXASXASScanActionController.cpp [hunk 1]
```diff
@@ -68,8 +70,9 @@ AMAction3* BioXASXASScanActionController::createInitializationActions()
 	// Initialize the scaler.
 
 	AMSequentialListAction3 *scalerInitialization = 0;
-	CLSSIS3820Scaler *scaler = CLSBeamline::clsBeamline()->scaler();
+//	CLSSIS3820Scaler *scaler = CLSBeamline::clsBeamline()->scaler();
```
Comment (iainworkman) [inline source/acquaman/BioXAS/BioXASXASScanActionController.cpp]: Is the commented code regarding the scaler in this class still required?

Comment (helfrij) [inline source/acquaman/BioXAS/BioXASXASScanActionController.cpp]: I think I would prefer to keep them for now, until initializing the Zebra has been implemented.

File: source/actions3/actions/AMDetectorWaitForAcquisitionStateAction.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+void AMDetectorWaitForAcquisitionStateAction::startImplementation()
+{
+	if(!detector_ && AMBeamlineSupport::beamlineDetectorAPI())
+		detector_ = AMBeamlineSupport::beamlineDetectorAPI()->exposedDetectorByInfo(*(detectorWaitForAcquisitionStateInfo()->detectorInfo()));
+
+	if(!detector_) {
+		AMErrorMon::alert(this,
+						  AMDETECTORINITIALIZEACTION_NO_VALID_DETECTOR,
+						  QString("There was an error initializing the detector '%1', because the detector was not found. Please report this problem to the Acquaman developers.").arg(detectorWaitForAcquisitionStateInfo()->name()));
```
Comment (iainworkman) [inline source/actions3/actions/AMDetectorWaitForAcquisitionStateAction.cpp]: By the way, if you want github to align these continuation lines properly you can set **Align continuation lines** to **With Spaces** in:
Options >> C++ >> Code Style >> General
and
Options >> Text Editor >> Beahviour

File: source/actions3/actions/AMDetectorWaitForAcquisitionStateActionInfo.h [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+#define AMDETECTORWAITFORACQUISITIONSTATEACTIONINFO_H
+
+#include "actions3/AMActionInfo3.h"
+#include "dataman/info/AMDetectorInfo.h"
+
+/// Holds the info for an action that waits for a specific acquisition state transition in the detector.
+class AMDetectorWaitForAcquisitionStateActionInfo : public AMActionInfo3
+{
+	Q_OBJECT
+
```
Comment (iainworkman) [inline source/actions3/actions/AMDetectorWaitForAcquisitionStateActionInfo.h]: Is this missing the Q_PROPERTY calls for the detectorInfo and acquisitionState?

Comment (dretrex) [inline source/actions3/actions/AMDetectorWaitForAcquisitionStateActionInfo.h]: Haha, yup.

**Comment by iainworkman** [comment]
Looks good to me.
Time Estimate: 90 minutes

**Pull request merged**